coreutils_core/src/mktemp.rs

```diff
@@ -0,0 +1,88 @@
+//! Module with wrappers for libc mkstemp(3), mkdtemp(3)
+
+use std::{
+    fmt::{self, Display},
+    fs::File,
+    io::Error,
+    os::unix::io::FromRawFd,
+};
+
+#[derive(Debug, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct MktempError {
+    err: String,
+}
+
+/// A struct that represents a mktemp(3) result.
+/// This includes the file created, and the path to that file.
+#[derive(Debug)]
+pub struct Mktemp {
+    pub file: File,
+    pub path: String,
```

Comment:
I think it's a better pattern to have fields private and have functions to get them, but some times are the right call, there is any specific reason to be public in this cases?

Reply:
The whole point is giving the caller access to the returned open file and its path, so it wouldn't be very meaningful to make them private and add getters, just extra work for no gain.

Reply:
Fair enought